source/tessellation/Delaunay.cpp

```diff
@@ -361,27 +363,19 @@ void Delaunay::update(const vector<Vector2D>& points, vector<Vector2D>
 
 namespace {
 
-	int Triplet<int>::* walk_condition(const vector<Vector2D>& cor,
-		const Triplet<int>& vertices,
-		size_t point)
+  boost::optional<size_t> walk_condition
+  (const vector<Vector2D>& cor,
+   const Triplet<int>& vertices,
+   size_t point)
 	{
-		if (orient2d(TripleConstRef<Vector2D>
-			(cor[static_cast<size_t>(vertices.first)],
-				cor[static_cast<size_t>(vertices.second)],
-				cor[point])) < 0)
-			return &Triplet<int>::first;
-		else if (orient2d(TripleConstRef<Vector2D>
-			(cor[static_cast<size_t>(vertices.second)],
-				cor[static_cast<size_t>(vertices.third)],
-				cor[point])) < 0)
-			return &Triplet<int>::second;
-		else if (orient2d(TripleConstRef<Vector2D>
-			(cor[static_cast<size_t>(vertices.third)],
-				cor[static_cast<size_t>(vertices.first)],
-				cor[point])) < 0)
-			return &Triplet<int>::third;
-		else
-			return nullptr;
+	  for(size_t i=0;i<3;++i){
+	    if (orient2d(TripleConstRef<Vector2D>
+			 (cor[static_cast<size_t>(vertices[i])],
+			  cor[static_cast<size_t>(vertices[(i+1)%3])],
+			  cor[point])) < 0)
```

Comment:
this does not return the same as the old function, this just returns an index

Reply:
Is it fixed that the return value is used differently?

Reply:
What difference does it make? You get the same functionality and the code is safer because I eliminated raw pointers